cpp/memilio/io/README.md

```diff
@@ -4,7 +4,24 @@ This directory contains utilities for reading and writing data from and to files
 
 ## The Serialization framework
 
-## Main functions and types
+### Using serialization
+
+In the next sections we will explain how to implement serialization (both for types and formats), here we quickly show
+how to use it once it already is implemented for a type. Currently, there is support for the Json and a binary format,
+which can be used through the `serialize_json`/`deserialize_json` and `serialize_binary`/`deserialize_binary`,
+respectively. For example
+
+```cpp
+Foo foo{5};
+mio::IOResult<Json::Value> js_result = mio::serialize_json(foo);
+```
+```cpp
+Json::Value js_value;
+js_value["i"] = Json::Int(5);
+mio::IOResult<Foo> foo_result = mio::deserialize_json(js_value, mio::Tag<Foo>{});
+```
```

Comment:
I think these examples early in the doc should only show the `read_json` and `write_json` functions that directly write to files. I think users are generally not expected to interact with `Json::Value`. Maybe the other versions can be mentioned in a later section.